### ISSUE TITLE
TinyMCE 6 dropdown item line height

### ISSUE THREAD

**Issue by brionymckenzie** [umbraco/Umbraco-CMS#13614]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

11.0.0

### Bug summary

The formats dropdown uses font family, colour and size props from custom stylesheets but doesn't update line height so if you have a large font size option, it overflows into the next item in the list. 
![image](https://user-images.githubusercontent.com/120108741/209144098-9a001c16-767a-4521-8e3b-2b3be926dd08.png)

### Specifics

I've tried adding margin, padding props and important tags but nothing seems to affect the line height. 
It wasn't an issue in TinyMCE 4:
![image](https://user-images.githubusercontent.com/120108741/209144825-67336b03-2a7f-48ec-98e5-6425e5566d33.png)

### Steps to reproduce

Create a rich text editor prop.
Configure style_formats with some options for different size headings, including some very large ones.
Apply a custom stylesheet with styles for the headings.

### Expected result / actual result

I expected the line height to update with the font size, but it stays fixed at 24px.

---
_This item has been added to our backlog [AB#25610](https://dev.azure.com/umbraco/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/25610)_

**Comment by Zeegaan** [comment]
Hey @brionymckenzie thanks for reporting this, this does indeed seem like a regression issue, and as you mention is probably caused by the upgrade to tiny MCE 6 😅 Do you per chance have some code snippets of what you've configured the style_formats with so i can test this ? 🤔

**Comment by brionymckenzie** [comment]
Hey, thanks for looking at this! 
Here's an example of what I've been using for the style_formats:
```
[
    {
        "title": "Sizes",
        "items": [
            {
                "title": "Heading Alpha",
                "selector": "p,span,h1,h2,h3,h4,h5,h6,a,ul,li,tr,td",
                "classes": "heading heading--alpha"
            },
            {
                "title": "Heading Bravo",
                "selector": "p,span,h1,h2,h3,h4,h5,h6,a,ul,li,tr,td",
                "classes": "heading heading--bravo"
            },
            {
                "title": "Heading Charlie",
                "selector": "p,span,h1,h2,h3,h4,h5,h6,a,ul,li,tr,td",
                "classes": "heading heading--charlie"
            },
            {
                "title": "Heading Delta",
                "selector": "p,span,h1,h2,h3,h4,h5,h6,a,ul,li,tr,td",
                "classes": "heading heading--delta"
            },
            {
                "title": "Heading Echo",
                "selector": "p,span,h1,h2,h3,h4,h5,h6,a,ul,li,tr,tda",
                "classes": "heading heading--echo"
            },
            {
                "title": "Heading Foxtrot",
                "selector": "p,span,h1,h2,h3,h4,h5,h6,a,ul,li,tr,td",
                "classes": "heading heading--foxtrot"
            },
            {
                "title": "Body Large",
                "selector": "p,span,h1,h2,h3,h4,h5,h6,a,ul,li,tr,td",
                "classes": "text text--large"
            },
            {
                "title": "Body Default",
                "selector": "p,span,h1,h2,h3,h4,h5,h6,a,ul,li,tr,td",
                "classes": "text text--default"
            },
            {
                "title": "Body Small",
                "selector": "p,span,h1,h2,h3,h4,h5,h6,a,ul,li,tr,td",
                "classes": "text text--small"
            }
        ]
    }
]
```

And here's the CSS:
```
@import url('https://fonts.googleapis.com/css2?family=Atkinson+Hyperlegible:wght@400;700&family=Poppins:wght@400;600;700&display=swap');

:root {
    --font-family-default: "Atkinson Hyperlegible", sans-serif;
    --font-family-heading: "Poppins", sans-serif;
    --x-body-line-height: 1.5;
}

body {
    font-size: 1rem;
    font-family: var(--font-family-default);
}

@media (min-width: 768px) {
    body {
        font-size: 1.19rem;
    }
}

h1,
h2,
h3,
h4,
h5,
h6 {
    font-family: var(--font-family-heading);
    font-weight: 400;
    margin: 0 0 1rem;
}

a {
    text-decoration: none;
}

.heading {
    font-family: var(--font-family-heading);
    font-weight: 400;
    line-height: calc(var(--x-body-line-height) * 0.9);
    margin: 0 0 1rem;
}

.heading--alpha {
    font-size: 2.58rem;
}

@media (min-width: 768px) {
    .heading--alpha {
        font-size: 3.0702rem;
    }
}

.heading--bravo {
    font-size: 2.16rem;
}

@media (min-width: 768px) {
    .heading--bravo {
        font-size: 2.5704rem;
    }
}

.heading--charlie {
    font-size: 1.79rem;
}

@media (min-width: 768px) {
    .heading--charlie {
        font-size: 2.1301rem;
    }
}

.heading--delta {
    font-size: 1.48rem;
}

@media (min-width: 768px) {
    .heading--delta {
        font-size: 1.7612rem;
    }
}

.heading--echo {
    font-size: 1.21rem;
}

@media (min-width: 768px) {
    .heading--echo {
        font-size: 1.4399rem;
    }
}

.heading--foxtrot {
    font-size: 1rem;
}

@media (min-width: 768px) {
    .heading--foxtrot {
        font-size: 1.19rem;
    }
}

.text {
    font-family: var(--font-family-default);
}

.text--large {
    font-size: 1.21rem;
}

@media (min-width: 768px) {
    .text--large {
        font-size: 1.4399rem;
    }
}

.text--default {
    font-size: 1rem;
}

@media (min-width: 768px) {
    .text--default {
        font-size: 1.19rem;
    }
}

.text--small {
    font-size: 0.85rem;
}

@media (min-width: 768px) {
    .text--small {
        font-size: 1.0115rem;
    }
}
```

**Comment by nikolajlauridsen** [comment]
Hey @brionymckenzie 

just had a go at testing this, and can confirm that this seems to be an issue with the new TinyRTE, I managed to reproduce it with a simpler stylesheet which I'll include in this comment for future help 😄 

End result:
![image](https://user-images.githubusercontent.com/16456704/210567575-1055f73a-d489-4dc5-882f-45fbdc834a69.png)


Stylesheet:

```css


/**umb_name:BIG*/
h1 {
	font-size:50pt;
}

/**umb_name:Bigger*/
h1 {
	font-size:75pt;
}

/**umb_name:Biggest*/
h1 {
	font-size:100pt;
}

/**umb_name:Blue*/
.blue {
	color: blue;
}

```

**Comment by iOvergaard** [comment]
So I tested this on Stackblitz with a standard TinyMCE and it looks like they hardcode the line-height to 24px, and even if you try and override it they do not pick it up in the dropdown menu (albeit it works in the content area). It doesn't look like something we can fix on our end:

![image](https://user-images.githubusercontent.com/752371/213204866-8e3cdccb-506a-4a05-ba16-66098d9a90a7.png)

Even their "default" content stylesheet has issues with line-height almost clipping:

![image](https://user-images.githubusercontent.com/752371/213205567-9a8a8f60-1a38-433f-b755-dd97bf970f57.png)

I'm afraid that the recommendation, for the time being, is not to set the font-size too high. I am trying to see if there is an open issue at Tiny.cloud or if we can report it.

It looks like tinymce/tinymce#6284 could potentially be the cause of this.

**Comment by iOvergaard** [comment]
This upstream issue looks like it fits the bill: tinymce/tinymce#6431

**Comment by iOvergaard** [comment]
I think we will try the fix suggested in the aforementioned issue. I made a PR for it: #13700 

If it passes review, it will be shipped with 11.2.0.

Sorry for comment spam!

**Comment by iOvergaard** [comment]
The fix has been merged and will be available in 11.2.0!
